web/packages/teleport/src/Discover/SelectResource/SelectResource.story.tsx

```diff
@@ -97,7 +97,15 @@ const Provider = ({
   const ctx = createTeleportContext({ customAcl: customAcl });
   const updatePreferences = () => Promise.resolve();
   const preferences: UserPreferences = makeDefaultUserPreferences();
-  preferences.onboard = { preferredResources: resources };
+  preferences.onboard = {
+    preferredResources: resources,
+    marketingParams: {
+      campaign: '',
+      source: '',
+      medium: '',
+      intent: '',
+    },
+  };
```

Comment:
Could we modify only the property that differs from the default preferences?
```suggestion
  preferences.onboard.preferredResources = resources
```

I would do it in other places as well.